pkg/bridge/ebtables/ebtables.go

```diff
@@ -0,0 +1,180 @@
+// Copyright 2017 CNI authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ebtables
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	osexec "os/exec"
+	"strconv"
+	"strings"
+	"syscall"
+
+	utilexec "github.com/containernetworking/plugins/pkg/exec"
+	"github.com/containernetworking/plugins/pkg/utils/sysctl"
+)
+
+const (
+	sysctlBridgeCallIPTables = "net/bridge/bridge-nf-call-iptables"
+)
+
+type ExitError interface {
+	Error() string
+	ExitStatus() int
+}
+
+func (e *Error) Error() string {
+	return fmt.Sprintf("exit status %v: %v", e.ExitStatus(), e.msg)
+}
+
+func (e *Error) ExitStatus() int {
+	return e.Sys().(syscall.WaitStatus).ExitStatus()
+}
+
+type Error struct {
+	*osexec.ExitError
+	msg string
+}
+
+var _ ExitError = &Error{}
+
+type EBTables struct {
+	path string
+	exec utilexec.Interface
+}
+
+func New(exec utilexec.Interface) (*EBTables, error) {
+	path, err := exec.LookPath("ebtables")
+	if err != nil {
+		return nil, err
+	}
+
+	ebt := EBTables{
+		path: path,
+		exec: exec,
+	}
+
+	modprobe, err := ebt.exec.LookPath("modprobe")
+	if err != nil {
+		return nil, err
+	}
+
+	var _, stderr bytes.Buffer
+	cmd := ebt.exec.Command(modprobe, "br-netfilter")
+	cmd.SetStderr(&stderr)
+
+	if err := cmd.Run(); err != nil {
+		return nil, &Error{(err.(*osexec.ExitError)), stderr.String()}
+	}
+
+	// Allow iptables to see bridged traffic, do this best effort.
+	sysctl.Sysctl(sysctlBridgeCallIPTables, "1")
+
```

Comment:
here also

Reply:
Also, just an implementation question, why is this set here in the plugin and not in the proxier that actually uses it?